### PR TITLE
*: opentracing for Execute,ParseSQL,Compile,runStmt

### PR DESCRIPTION
**Pull request by tiancaiamao**
Now we get here:
image

Next time, I'll trace some executor information such as coprocessor read operation...

### CODE REVIEW

**Comment by ngaut** [comment]
👍

**Comment by zz-jason** [comment]
LGTM

**Comment by zz-jason** [comment]
@zimulala @coocood PTAL

File: tidb.go [hunk 1]
```diff
@@ -162,10 +166,10 @@ func runStmt(ctx context.Context, s ast.Statement) (ast.RecordSet, error) {
 	if !se.sessionVars.InTxn() {
 		if err != nil {
 			log.Info("RollbackTxn for ddl/autocommit error.")
-			err1 := se.RollbackTxn()
+			err1 := se.RollbackTxn(ctx1)
 			terror.Log(errors.Trace(err1))
 		} else {
-			err = se.CommitTxn()
+			err = se.CommitTxn(ctx.GoCtx())
```
Comment (coocood) [inline tidb.go]: Why not use `ctx1`?

File: session.go [hunk 2]
```diff
@@ -846,7 +869,11 @@ func (s *session) Txn() kv.Transaction {
 
 func (s *session) NewTxn() error {
 	if s.txn != nil && s.txn.Valid() {
-		err := s.CommitTxn()
+		ctx := s.goCtx
+		if ctx == nil {
```
Comment (coocood) [inline session.go]: When will `ctx` be nil?

File: session.go [hunk 3]
```diff
@@ -66,8 +67,8 @@ type Session interface {
 	AffectedRows() uint64                        // Affected rows by latest executed stmt.
 	Execute(sql string) ([]ast.RecordSet, error) // Execute a sql statement.
 	String() string                              // String is used to debug.
-	CommitTxn() error
-	RollbackTxn() error
+	CommitTxn(goctx.Context) error
```
Comment (coocood) [inline session.go]: Can we not adding `goctx.Context`, always use `Session.GoCtx` in implementation?

File: session.go (hunk 2 above)
Comment (tiancaiamao) [reply session.go]: Normal cases won't go here, but in the testing `ctx` maybe not set.

File: session.go (hunk 3 above)
Comment (tiancaiamao) [reply session.go]: It's better to add this argument, and avoid store ctx in the object if possible.
For example, session start a txn and store the txn to `session.GoCtx`,
then call function `F(ctx)`,
`F(ctx)` call `CommitTxn()... span parent of `CommitTxn()` should be `F(ctx)`, not the `session.GoCtx`

**Pull request merged**